frontend/app/src/components/defi/Lending.vue

```diff
@@ -117,7 +154,42 @@ export default class Lending extends Vue {
   accountGain!: (address: string) => BigNumber;
   accountGainUsdValue!: (address: string) => BigNumber;
   dsrHistory!: AccountDSRMovement[];
+  dsrHistoryByAddress!: DSRHistory;
   isTaskRunning!: (type: TaskType) => boolean;
+  filteredAccounts: GeneralAccount[] = [];
+
+  async mounted() {
+    await this.$store.dispatch('balances/fetchDSRBalances');
+    if (this.premium) {
+      await this.$store.dispatch('balances/fetchDSRHistory');
+    }
+  }
+
+  get lendingAddresses(): Account[] {
+    let addresses: Account[] = [];
+
+    // join addresses from dsrHistory and dsrBalances
+    const dsrHistoryAddresses = this.dsrHistory.map(dsrEvent => {
+      return { chain: 'ETH', address: dsrEvent.address } as Account;
+    });
+
+    const dsrBalanceAddresses = this.dsrBalances.map(dsrBalance => {
+      return { chain: 'ETH', address: dsrBalance.address } as Account;
+    });
+
+    addresses = [...dsrHistoryAddresses, ...dsrBalanceAddresses];
+
+    // remove duplicates
+    addresses = addresses.filter(
+      (value, index, array) =>
+        array.findIndex(
+          filterBy =>
+            filterBy.chain === value.chain && filterBy.address === value.address
```

Comment:
is the chain really needed at the moment?

Reply:
Not needed now but it doesn't hurt I guess? I wasn't sure if it was possible for addresses to not be unique in general (e.g. if an address could potentially exist on multiple chains).